src/api/spec/models/owner_spec.rb

```diff
@@ -0,0 +1,120 @@
+        create(:relationship_package_user, package: package, user: user, role: Role.find_by_title('bugowner'))
+        user.update_attributes(state: :locked)
+
+        subject = Owner.search({ devel: false, filter: 'bugowner' }, nil).first
+        expect(subject.rootproject).to eq('home:Iggy')
+        expect(subject.project).to eq('home:Iggy')
+        expect(subject.package).to eq('TestPack')
+        expect(subject.users).to be_nil
+      end
+
+      it 'respects User.owner' do
+        create(:relationship_package_user, package: package, user: user, role: Role.find_by_title('bugowner'))
+        user.update_attributes(owner: develuser)
+
+        subject = Owner.search({ devel: false, filter: 'bugowner' }, nil)
+        expect(subject).to eq([])
+
+        develuser.update_attributes(state: :locked)
+
+        subject = Owner.search({ devel: false, filter: 'bugowner' }, nil).first
+        expect(subject.rootproject).to eq('home:Iggy')
+        expect(subject.project).to eq('home:Iggy')
+        expect(subject.package).to eq('TestPack')
+      end
+    end
+
+    context 'in maintenance projects' do
+      let(:project) { Project.find_by(name: 'home:Iggy') }
+      let!(:project_kind) { project.update_attributes(kind: 'maintenance_release') }
+      let(:other_user) { create(:confirmed_user, login: 'hans') }
+
+      # A package with bugowner develuser
+      let(:package_42) { create(:package, name: 'package.42', project: project) }
+      # FIXME: bugowner should be a transitive argument to the package factory
+      let!(:bugowner) { create(:relationship_package_user_as_bugowner, user: develuser, package: package_42) }
+
+      # A local link to package.42 with bugowner other_user
+      let(:package) { create(:package, name: 'package', project: project) }
+      # FIXME: package link should be a transitive argument to the package factory
+      let!(:package_link) { package.build_backend_package.update_attributes(links_to_id: package_42) }
+      let!(:other_bugowner) { create(:relationship_package_user_as_bugowner, user: other_user, package: package) }
+
+      # A patchinfo with bugowner maintenance_user
+      let(:patchinfo_42) { create(:package, name: 'patchinfo.42', project: project) }
+      # FIXME: package kind should be a transitive argument to the package factory
+      let!(:kind_patchinfo) { PackageKind.create(package_id: patchinfo_42.id, kind: 'patchinfo') }
+      let(:maintenance_user) { create(:confirmed_user, login: 'MaintenanceIggy') }
+      # FIXME: bugowner should be a transitive argument to the package factory
+      let!(:maintenance_bugowner) { create(:relationship_package_user_as_bugowner, user: maintenance_user, package: patchinfo_42) }
+
+      before do
+        allow(Backend::Api::Search).to receive(:binary).and_return(maintenance_collection)
+      end
+      it 'respects maintenance project suffixes' do
+        expect(Owner.search({}, 'package').first.users).to eq('bugowner'=>['hans'])
```

Comment:
this is supposed to test search for package not for 'package', right?

Reply:
scratch that - I didn't realize just how horrible this api is

Reply:
...